razorpay/client.py

```diff
@@ -121,7 +122,7 @@ def request(self, method, path, **options):
                                                  **options)
         if ((response.status_code >= HTTP_STATUS_CODE.OK) and
                 (response.status_code < HTTP_STATUS_CODE.REDIRECT)):
-            return response.json()
+            return response.json() if(len(response.text) > 0) else response.status_code
```

Comment:
By the looks of it, we will sometime return a json string and sometime a code? why are we doing it this way? Why not always return a json string.

Problem with this is, the caller will have to handle different type of responses.

Reply:
@razorpay-sanjib because while integrating [delete allowed payer](https://razorpay.com/docs/api/smart-collect-tpv/#delete-an-allowed-payer-account) its return http 204 status code thats why i that condition

Reply:
So this DELETE request, which returns no response in case of success. It’s just a status code 204. In other cases, if there are errors, we receive a json response.

One case we are returning a number(status code) and in other case we are returning a response (json). But if you see at the API, technically, incase of success it’s an empty response and in case of error, it sends json response, do the same. return a response, not a status code.

So here check if the status_code is 204. If yes return an empty response, which aligns with the API response.